core/shared/src/main/scala/fs2/async/async.scala

```diff
@@ -80,33 +78,32 @@ package object async {
     * an arbitrary number of subscribers. Each subscriber is guaranteed to
     * receive at least the initial `A` or last value published by any publisher.
     */
-  def topic[F[_]: Effect, A](initial: A)(implicit ec: ExecutionContext): F[mutable.Topic[F, A]] =
+  def topic[F[_]: Concurrent, A](initial: A)(
+      implicit ec: ExecutionContext): F[mutable.Topic[F, A]] =
     mutable.Topic(initial)
 
   /** Creates an empty `Promise[F, A]` */
-  def promise[F[_]: Effect, A](implicit ec: ExecutionContext): F[Promise[F, A]] = Promise.empty
+  def promise[F[_]: Concurrent, A]: F[Promise[F, A]] = Promise.empty
 
   /** Creates an initialized `SyncRef[F,A]`. */
   def refOf[F[_]: Sync, A](a: A): F[Ref[F, A]] = Ref[F, A](a)
 
   /** Like `traverse` but each `G[B]` computed from an `A` is evaluated in parallel. */
   def parallelTraverse[F[_], G[_], A, B](fa: F[A])(
-      f: A => G[B])(implicit F: Traverse[F], G: Effect[G], ec: ExecutionContext): G[F[B]] =
+      f: A => G[B])(implicit F: Traverse[F], G: Concurrent[G], ec: ExecutionContext): G[F[B]] =
     F.traverse(fa)(f.andThen(start[G, B])).flatMap(F.sequence(_))
 
   /** Like `sequence` but each `G[A]` is evaluated in parallel. */
   def parallelSequence[F[_], G[_], A](
-      fga: F[G[A]])(implicit F: Traverse[F], G: Effect[G], ec: ExecutionContext): G[F[A]] =
+      fga: F[G[A]])(implicit F: Traverse[F], G: Concurrent[G], ec: ExecutionContext): G[F[A]] =
     parallelTraverse(fga)(identity)
 
   /**
     * Begins asynchronous evaluation of `f` when the returned `F[F[A]]` is
     * bound. The inner `F[A]` will block until the result is available.
     */
-  def start[F[_], A](f: F[A])(implicit F: Effect[F], ec: ExecutionContext): F[F[A]] =
-    promise[F, Either[Throwable, A]].flatMap { p =>
-      fork(f.attempt.flatMap(p.complete)).as(p.get.flatMap(F.fromEither))
-    }
+  def start[F[_], A](f: F[A])(implicit F: Concurrent[F], ec: ExecutionContext): F[F[A]] =
```

Comment:
This will be eventually removed since we have it in cats effect, right?
Or do we want to keep a different version that `shifts`? and if so, why?

Reply:
It's needed for the implementation of `parallelTraverse` due to the thread shifting here. If we just used `Concurrent[F].start` in `parallelTraverse`, we won't get as much concurrency as folks expect. Perhaps we should deprecate `parallelTraverse` in favor of the new `Parallel` support in cats-effect though?

Reply:
I think we should. I like that fs2 has its own concurrency scheme, but I'm the way I look at `parallelTraverse` here is just parallelism, in which case using `Parallel` is probably enough.
When you need more complicated concurrency, you should go with `join`.

So I'm slightly leaning toward removing both `start` and `parallel*` in favour of the ones in cats-effect. Not a strong opinion though.